Mathlib/CategoryTheory/Functor/KanExtension/Pointwise.lean

```diff
@@ -0,0 +1,191 @@
+    (fun G => by
+      suffices ∀ (m₁ m₂ : E ⟶ G), m₁ = m₂ by intros; apply this
+      intro m₁ m₂
+      ext Y
+      apply (h Y).hom_ext
+      intro X
+      have eq₁ := congr_app (StructuredArrow.w m₁) X.left
+      have eq₂ := congr_app (StructuredArrow.w m₂) X.left
+      dsimp at eq₁ eq₂ ⊢
+      simp only [assoc, NatTrans.naturality]
+      rw [reassoc_of% eq₁, reassoc_of% eq₂])
+
+lemma IsPointwiseLeftKanExtension.isLeftKanExtension :
+    E.right.IsLeftKanExtension E.hom where
+  nonempty_isUniversal := ⟨h.isUniversal⟩
+
+lemma IsPointwiseLeftKanExtension.hasLeftKanExtension :
+    HasLeftKanExtension L F :=
+  have := h.isLeftKanExtension
+  HasLeftKanExtension.mk E.right E.hom
+
+end LeftExtension
+
+section
+
+variable [HasPointwiseLeftKanExtension L F]
+
+/-- The constructed pointwise left Kan extension when `HasPointwiseLeftKanExtension L F` holds. -/
+@[simps]
+noncomputable def pointwiseLeftKanExtension : D ⥤ H where
+  obj Y := colimit (CostructuredArrow.proj L Y ⋙ F)
+  map {Y₁ Y₂} f :=
+    colimit.desc (CostructuredArrow.proj L Y₁ ⋙ F)
+      (Cocone.mk (colimit (CostructuredArrow.proj L Y₂ ⋙ F))
+        { app := fun g => colimit.ι (CostructuredArrow.proj L Y₂ ⋙ F)
+            ((CostructuredArrow.map f).obj g)
+          naturality := fun g₁ g₂ φ => by
+            simpa using colimit.w (CostructuredArrow.proj L Y₂ ⋙ F)
+              ((CostructuredArrow.map f).map φ) })
+  map_id Y := colimit.hom_ext (fun j => by
+    dsimp
+    simp only [colimit.ι_desc, comp_id]
+    congr
+    apply CostructuredArrow.map_id)
+  map_comp {Y₁ Y₂ Y₃} f f' := colimit.hom_ext (fun j => by
+    dsimp
+    simp only [colimit.ι_desc, colimit.ι_desc_assoc, comp_obj, CostructuredArrow.proj_obj]
+    congr 1
+    apply CostructuredArrow.map_comp)
+
+/-- The unit of the constructed pointwise left Kan extension when
+`HasPointwiseLeftKanExtension L F` holds. -/
+@[simps]
+noncomputable def pointwiseLeftKanExtensionUnit : F ⟶ L ⋙ pointwiseLeftKanExtension L F where
+  app X := colimit.ι (CostructuredArrow.proj L (L.obj X) ⋙ F)
+    (CostructuredArrow.mk (𝟙 (L.obj X)))
+  naturality {X₁ X₂} f:= by
+    simp only [comp_obj, pointwiseLeftKanExtension_obj, comp_map,
+      pointwiseLeftKanExtension_map, colimit.ι_desc, CostructuredArrow.map_mk]
+    rw [id_comp]
+    let φ : CostructuredArrow.mk (L.map f) ⟶ CostructuredArrow.mk (𝟙 (L.obj X₂)) :=
+      CostructuredArrow.homMk f
+    exact colimit.w (CostructuredArrow.proj L (L.obj X₂) ⋙ F) φ
+
+/-- The functor `pointwiseLeftKanExtension L F` is a pointwise left Kan
+extension of `F` along `L`. -/
+noncomputable def pointwiseLeftKanExtensionIsPointwiseLeftKanExtension :
+    (LeftExtension.mk _ (pointwiseLeftKanExtensionUnit L F)).IsPointwiseLeftKanExtension :=
+  fun X => IsColimit.ofIsoColimit (colimit.isColimit _) (Cocones.ext (Iso.refl _) (fun j => by
+    dsimp
+    simp only [comp_id, colimit.ι_desc, CostructuredArrow.map_mk]
+    congr 1
+    rw [id_comp]
+    rfl))
```

Comment:
`rfl` after `rw` to finish off a proof is generally a code smell as it indicates that there may be missing API, but it's not necessary here.
```suggestion
    rw [id_comp, ← CostructuredArrow.eq_mk]
```